tests/js/server/aql/aql-optimizer-rule-late-document-materialization-arangosearch-2.js

```diff
@@ -0,0 +1,334 @@
+      db._drop(cn);
+      db._drop(cn1);
+
+      let c = db._create(cn, { numberOfShards: 3 });
+      let c2 = db._create(cn1, { numberOfShards: 3 });
+
+      db._createView(vn, "arangosearch", {
+        links: {
+          [cn] : { includeAllFields: true, analyzers : [ "identity" ],
+                   fields : { str : { "analyzers" : [ "text_en" ] }}, },
+          [cn1] : { includeAllFields: true, analyzers : [ "identity" ],
+                    fields : { str : { "analyzers" : [ "text_en" ] }}}
+        }});
+      db._createView(svn, "arangosearch", {
+        consolidationIntervalMsec: 5000,
+        primarySort: [{"field": "value", "direction": "asc"}, {"field": "foo", "direction": "desc"}],
+        links: {
+          [cn] : { includeAllFields: true },
+          [cn1] : { includeAllFields: true }
+        }});
+
+      c.save({ _key: 'c0',  str: 'cat', foo: 'foo0', value: 0 });
+      c2.save({_key: 'c_0', str: 'cat', foo: 'foo1', value: 10 });
+      c.save({ _key: 'c1',  str: 'cat', foo: 'foo2', value: 1 });
+      c2.save({_key: 'c_1', str: 'cat', foo: 'foo3', value: 11 });
+      c.save({ _key: 'c2',  str: 'cat', foo: 'foo4', value: 2 });
+      c2.save({_key: 'c_2', str: 'cat', foo: 'foo5', value: 12 });
+      c.save({ _key: 'c3',  str: 'cat', foo: 'foo6', value: 3 });
+      c2.save({_key: 'c_3', str: 'cat', foo: 'foo7', value: 13 });
+
+      // trigger view sync
+      db._query("FOR d IN " + vn + " OPTIONS { waitForSync: true } RETURN d");
+      db._query("FOR d IN " + svn + " OPTIONS { waitForSync: true } RETURN d");
+    },
+
+    tearDownAll : function () {
+      try { db._dropView(vn); } catch(e) {}
+      try { db._dropView(svn); } catch(e) {}
+      try { db._drop(cn); } catch(e) {}
+      try { db._drop(cn1); } catch(e) {}
+    },
+    testNotAppliedDueToNoPrimarySort() {
+      let query = "FOR d IN " + vn  + " SEARCH d.value IN [1, 2] SORT d.foo DESC LIMIT 10 RETURN d";
+      let plan = AQL_EXPLAIN(query).plan;
+      assertEqual(-1, plan.rules.indexOf(ruleName));
+    },
+    testNotAppliedDueToSortElimination() {
+      let query = "FOR d IN " + svn + " LET c = d.value + RAND() SORT d.value LIMIT 10 RETURN d";
+      let plan = AQL_EXPLAIN(query).plan;
+      assertEqual(-1, plan.rules.indexOf(ruleName));
+    },
+    testNotAppliedDueToNoSort() {
+      let query = "FOR d IN " + svn + " LIMIT 10 RETURN d";
+      let plan = AQL_EXPLAIN(query).plan;
+      assertEqual(-1, plan.rules.indexOf(ruleName));
+    },
+    testNotAppliedDueToUsedInInnerSort() {
+      let query = "FOR d IN " + svn + " SORT NOOPT(d.str) SORT d.value DESC LIMIT 10 RETURN d";
+      let plan = AQL_EXPLAIN(query).plan;
+      assertEqual(-1, plan.rules.indexOf(ruleName));
+    },
+    testNotAppliedDueToNoLimit() {
+      let query = "FOR d IN " + svn + " SORT d.value DESC RETURN d";
+      let plan = AQL_EXPLAIN(query).plan;
+      assertEqual(-1, plan.rules.indexOf(ruleName));
+    },
+    testQueryResultsWithFilter() {
```

Comment:
Let`s also add test for late materialization NOT applied if variable is used in subquery (as we added such execution branch in optimizer rule for views).

Reply:
Done